src/modules/fw_wind_estimator/FixedwingWindEstimator.cpp

```diff
@@ -0,0 +1,307 @@
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "FixedwingWindEstimator.hpp"
+
+using namespace time_literals;
+using namespace matrix;
+
+using math::constrain;
+using math::interpolate;
+using math::radians;
+
+FixedwingWindEstimator::FixedwingWindEstimator(bool vtol) :
+	ModuleParams(nullptr),
+	ScheduledWorkItem(MODULE_NAME, px4::wq_configurations::nav_and_controllers),
+	_loop_perf(perf_alloc(PC_ELAPSED, MODULE_NAME": cycle"))
+{
+	/* fetch initial parameter values */
+	parameters_update();
+
+}
+
+FixedwingWindEstimator::~FixedwingWindEstimator()
+{
+	perf_free(_loop_perf);
+}
+
+bool
+FixedwingWindEstimator::init()
+{
+	if (!_vehicle_angular_velocity_sub.registerCallback()) {
+		PX4_ERR("callback registration failed");
+		return false;
+	}
+
+	return true;
+}
+
+int
+FixedwingWindEstimator::parameters_update()
+{
+
+	_mass = _param_fw_w_mass.get();
+	_stall_airspeed = _param_fw_airspd_stall.get();
+
+	// Get Aerodynamic coefficients
+	_wing_area = _param_fw_w_area.get();
+	_C_B1 = _param_fw_w_c_b1.get();
+	_C_A0 = _param_fw_w_c_a0.get();
+	_C_A1 = _param_fw_w_c_a1.get();
+
+	return PX4_OK;
+}
+
+void
+FixedwingWindEstimator::vehicle_land_detected_poll()
+{
+	if (_vehicle_land_detected_sub.updated()) {
+		vehicle_land_detected_s vehicle_land_detected {};
+
+		if (_vehicle_land_detected_sub.copy(&vehicle_land_detected)) {
+			_landed = vehicle_land_detected.landed;
+		}
+	}
+}
+
+void FixedwingWindEstimator::airspeed_poll()
+{
+	airspeed_validated_s airspeed_validated;
+
+	if (_airspeed_validated_sub.update(&airspeed_validated)) {
+		_calibrated_airspeed = math::max(0.5f, airspeed_validated.calibrated_airspeed_m_s);
+		_true_airspeed = math::max(0.5f, airspeed_validated.true_airspeed_m_s);
+	}
+}
+
+
+void
+FixedwingWindEstimator::vehicle_attitude_poll()
+{
+	if (_vehicle_attitude_sub.update(&_vehicle_attitude)) {
+		// get rotation between NED frames
+		_attitude = Quatf(_vehicle_attitude.q);
+	}
+}
+
+void
+FixedwingWindEstimator::vehicle_local_position_poll()
+{
+	vehicle_local_position_s vehicle_local_position;
+
+	if (_vehicle_local_position_sub.update(&vehicle_local_position)) {
+		// get rotation between NED frames
+		_local_velocity = Vector3f(vehicle_local_position.vx, vehicle_local_position.vy, vehicle_local_position.vz);
+	}
+}
+
+void
+FixedwingWindEstimator::vehicle_acceleration_poll()
+{
+	//vehicle_local_position_s pos;
+	///TODO: We should probably get it from the imu, not the local one?
+	vehicle_acceleration_s vehicle_acceleration;
+
+	if (_vehicle_acceleration_sub.update(&vehicle_acceleration)) {
+		Dcmf R_ib(_attitude);
+		_acceleration = R_ib * Vector3f(vehicle_acceleration.xyz);
```

Comment:
```suggestion
		_acceleration = _attitude.rotateVector(vehicle_acceleration.xyz);
```
This is faster when rotating a single vector.